include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld

```diff
@@ -266,36 +266,9 @@ SECTIONS
     _app_smem_rom_start = LOADADDR(_APP_SMEM_SECTION_NAME);
 #endif  /* CONFIG_USERSPACE */
 
-    SECTION_DATA_PROLOGUE(_BSS_SECTION_NAME,(NOLOAD), BSS_ALIGN)
-    {
-        /*
-         * For performance, BSS section is assumed to be 4 byte aligned and
-         * a multiple of 4 bytes
-         */
-        . = ALIGN(4);
-        __bss_start = .;
-        __kernel_ram_start = .;
-
-        *(.bss)
-        *(".bss.*")
-        *(COMMON)
-        *(".kernel_bss.*")
-
-#ifdef CONFIG_CODE_DATA_RELOCATION
-#include <linker_sram_bss_relocate.ld>
-#endif
-
-        /*
-         * As memory is cleared in words only, it is simpler to ensure the BSS
-         * section ends on a 4 byte boundary. This wastes a maximum of 3 bytes.
-         */
-        __bss_end = ALIGN(4);
-    } GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
-
-#include <zephyr/linker/common-noinit.ld>
-
     SECTION_DATA_PROLOGUE(_DATA_SECTION_NAME,,)
     {
+        __kernel_ram_start = .;
```

Comment:
can you move this outside the data section?

Reply:
I can test that, sure. I've had a case before though where putting a marker outside a section using the "." operator produced the correct value for the marker, however, the following section's start didn't fit the marker due to alignment. The trouble was that the following section aligned itself to a location *before* the marker outside of the section. That was... weird. I'll see if I still get the same values for the markers if the start marker is outside of the section.

Reply:
Done. I looked it up, the issue I was referring to in my last comment was #41513, btw. In that case, the issue was having a '.'-based marker outside of a region, not outside of a section. So moving `__kernel_ram_start` outside of the data section produces valid results.